commitizen/commands/commit.py

```diff
@@ -63,10 +64,18 @@ def prompt_commit_questions(self) -> str:
 
     def __call__(self):
         dry_run: bool = self.arguments.get("dry_run")
+        write_message_to_file = self.arguments.get("write_message_to_file")
 
         if git.is_staging_clean() and not dry_run:
             raise NothingToCommitError("No files added to staging!")
 
+        if write_message_to_file is not None:
+            if not isinstance(write_message_to_file, str):
+                raise NotAllowed(
+                    "Commit message file name is broken.\n"
+                    "Check the flag `--write-message-to-file` in the terminal"
+                )
```

Comment:
`smart_open` in line 89 does raise an error if the path points to an existing directory but, considering that the user had to go through answering the questions at that point, it makes sense to already raise an error here. does the cli library support `pathlib.Path` as a type?

if not, i'd add a check like this:

```suggestion
        if write_message_to_file is not None:
            if not isinstance(write_message_to_file, str):
                raise NotAllowed(
                    "Commit message file name is broken.\n"
                    "Check the flag `--write-message-to-file` in the terminal"
                )
    
            if os.path.isdir(write_message_to_file):
                raise NotAllowed(f"{write_message_to_file} is a directory")
```

Reply:
it seems like you're using `argparse` for [woile/decli](https://github.com/woile/decli) so we could also change the type of `write_message_to_file` to `pathlib.Path` and let `argparse` do the argument validation. then this would simplify to:

```suggestion
        if write_message_to_file is not None and write_message_to_file.is_dir():
            raise NotAllowed(f"{write_message_to_file} is a directory")
```

Reply:
I like the idea of using `Path` 👍🏻